File: client/src/app/modules/invoicing/services/infogreffe.invoice.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { InfogreffeInvoice } from '../../invoicing/model/InfogreffeInvoice';
import { Provision } from '../../quotation/model/Provision';

@Injectable({
  providedIn: 'root'
})
export class InfogreffeInvoiceService extends AppRestService<InfogreffeInvoice>{

  constructor(http: HttpClient) {
    super(http, "invoicing");
  }

  getInfogreffeInvoices() {
    return this.getList(new HttpParams(), "infogreffe-invoices");
  }

  getInfogreffeInvoicesByCustomerReference(customerReference: string) {
    return this.getList(new HttpParams().set("customerReference", customerReference), "infogreffe-invoices/search");
  }

  importCsv(csv: string) {
    return this.postItem(new HttpParams(), "infogreffe-invoice/import", csv);
  }

  createInvoiceFromGreffeInvoice(greffeInvoice: InfogreffeInvoice, provision: Provision) {
    return this.get(new HttpParams().set("infogreffeInvoiceId", greffeInvoice.id).set("provisionId", provision.id), "infogreffe-invoice/create");
  }
}

File: src/main/java/com/jss/osiris/modules/invoicing/service/InfogreffeInvoiceService.java
package com.jss.osiris.modules.invoicing.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisClientMessageException;
import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.libs.exception.OsirisValidationException;
import com.jss.osiris.modules.invoicing.model.InfogreffeInvoice;
import com.jss.osiris.modules.invoicing.model.Invoice;
import com.jss.osiris.modules.quotation.model.Provision;

public interface InfogreffeInvoiceService {
    public List<InfogreffeInvoice> getInfogreffeInvoices();

    public List<InfogreffeInvoice> getInfogreffeInvoicesByCustomerReference(String customerReference);

    public InfogreffeInvoice getInfogreffeInvoice(Integer id);

    public InfogreffeInvoice addOrUpdateInfogreffeInvoice(InfogreffeInvoice infogreffeInvoice);

    public Boolean importInfogreffeInvoices(String csv)
            throws OsirisException, OsirisClientMessageException, OsirisValidationException;

    public Invoice generateInvoiceFromProvisionAndGreffeInvoice(InfogreffeInvoice greffeInvoice, Provision provision)
            throws OsirisException, OsirisClientMessageException, OsirisValidationException;
}
